stress: failed test in cockroach/storage/storage.test: TestSkipLargeReplicaSnapshot

Binary: cockroach/static-tests.tar.gz sha: https://github.com/cockroachdb/cockroach/commits/2375e5d9b2b00b61418fa0b511035e7a10889e3c

Stress build found a failed test:

```
=== RUN   TestSkipLargeReplicaSnapshot
I161004 05:07:42.198870 1439 gossip/gossip.go:234  initial resolvers: []
W161004 05:07:42.198895 1439 gossip/gossip.go:1045  no resolvers found; use --join to specify a connected node
I161004 05:07:42.198942 1439 gossip/gossip.go:268  NodeID set to 1
I161004 05:07:42.199050 1439 storage/engine/rocksdb.go:360  opening in memory rocksdb instance
I161004 05:07:42.200406 1439 gossip/gossip.go:268  NodeID set to 1
I161004 05:07:42.200520 1439 gossip/gossip.go:268  NodeID set to 1
I161004 05:07:42.203336 1585 storage/replica_trigger.go:252  [s1,r1:/M{in-ax}] new range lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 900ms following replica {0 0 0} 1970-01-01 00:00:00 +0000 UTC 0s [physicalTime=1970-01-01 00:00:00 +0000 UTC]
I161004 05:07:42.777877 1439 storage/replica_raftstorage.go:443  [s1,r1:/M{in-ax}] generated snapshot b34b8e34 for range 1 at index 930 in 67.895µs.
I161004 05:07:43.855880 1439 storage/replica_raftstorage.go:443  [s1,r1:/M{in-ax}] generated snapshot 968cf6f3 for range 1 at index 2767 in 75.965µs.
I161004 05:07:43.855947 1439 util/stop/stopper.go:353  stop has been called, stopping or quiescing all running tasks
--- FAIL: TestSkipLargeReplicaSnapshot (1.67s)
	replica_raftstorage_test.go:97: snapshot of a very large range (3078608 / 67108864, needsSplit: false, exceeds snap limit: false) should fail but got <nil>
```

Run Details:

```
0 runs so far, 0 failures, over 5s
0 runs so far, 0 failures, over 10s
0 runs so far, 0 failures, over 15s
0 runs so far, 0 failures, over 20s
1 runs so far, 0 failures, over 25s
15 runs so far, 0 failures, over 30s
16 runs so far, 0 failures, over 35s
16 runs so far, 0 failures, over 40s
16 runs so far, 0 failures, over 45s
17 runs so far, 0 failures, over 50s
25 runs so far, 0 failures, over 55s
30 runs so far, 0 failures, over 1m0s
32 runs so far, 0 failures, over 1m5s
32 runs so far, 0 failures, over 1m10s
34 runs so far, 0 failures, over 1m15s
38 runs so far, 0 failures, over 1m20s
42 runs completed, 1 failures, over 1m21s
FAIL
```
Please assign, take a look and update the issue accordingly.

--- a/storage/replica_raftstorage_test.go
+++ b/storage/replica_raftstorage_test.go
@@ -59,13 +59,14 @@ func TestSkipLargeReplicaSnapshot(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	storeCfg := TestStoreConfig()
 	storeCfg.TestingKnobs.DisableSplitQueue = true
-	store, _, stopper := createTestStoreWithContext(t, &storeCfg)
-	defer stopper.Stop()
 
 	const snapSize = 1 << 20 // 1 MiB
 	cfg := config.DefaultZoneConfig()
 	cfg.RangeMaxBytes = snapSize
 	defer config.TestingSetDefaultZoneConfig(cfg)()
+
+	store, _, stopper := createTestStoreWithContext(t, &storeCfg)
+	defer stopper.Stop()
 
 	rep, err := store.GetReplica(rangeID)
 	if err != nil {

